scripts/browserstack.py

```diff
@@ -161,58 +161,51 @@ def uploadEspressoApk(espressoApkFile):
         return None
 
 
-def getAllTestClassNames():
-    testClassNames = [
-        # Hard coded tests.
-        "com.stripe.android.TestAuthorization",
-        "com.stripe.android.TestBrowsers",
-        "com.stripe.android.TestCustomers",
-        "com.stripe.android.addresselement.AddressElementTest",
-    ]
-
-    # Add LPM tests to testClassNames.
-    lpmFileNames = os.listdir(
-        "paymentsheet-example/src/androidTest/java/com/stripe/android/lpm"
-    )
-    for fileName in lpmFileNames:
-        if not fileName.endswith(".kt"):
-            continue
-        className = fileName[:-3]
-        testClassNames.append(f"com.stripe.android.lpm.{className}")
+def runFastScandir(dir, ext):
+    subfolders, files = [], []
 
-    return testClassNames
+    for f in os.scandir(dir):
+        if f.is_dir():
+            subfolders.append(f.path)
+        if f.is_file():
+            if os.path.splitext(f.name)[1].lower() in ext:
+                files.append(f.path)
 
+    for dir in list(subfolders):
+        sf, f = runFastScandir(dir, ext)
+        subfolders.extend(sf)
+        files.extend(f)
+
+    return subfolders, files
+
+
+def testFilesFromKotlinFiles(kotlinFiles):
+    testFiles = []
+
+    for file in kotlinFiles:
+        with open(file) as file_handle:
+            if 'import org.junit.Test' in file_handle.read():
+                testFiles.append(file)
+                continue
+
+    return testFiles
 
-def testShards(isNightly, testClassNames):
-    # We only have 25 parallel runs, and we want multiple PRs to run at the same time.
-    numberOfShards = 2.0 if isNightly else 10.0
-    testClassesPerShard = math.ceil(len(testClassNames) / numberOfShards)
-
-    shards = []
-    shardValues = []
-
-    for testClass in testClassNames:
-        shardValues.append(testClass)
-        if len(shardValues) == testClassesPerShard:
-            shards.append(
-                {
-                    "name": f"Shard {len(shards) + 1}",
-                    "strategy": "class",
-                    "values": shardValues,
-                },
-            )
-            shardValues = []
-
-    if len(shardValues) > 0:
-        shards.append(
-            {
-                "name": f"Shard {len(shards) + 1}",
-                "strategy": "class",
-                "values": shardValues,
-            },
-        )
 
-    return shards
+def classNamesFromTestFiles(testDirectory, testFileNames):
+    classNames = []
+
+    for fileName in testFileNames:
+        className = (fileName[len(testDirectory):][:-3]).replace('/', '.')
```

Comment:
do you think you could add some comments or a helper function to make it more obvious what this is doing?